pkg/interactive/docker.go

```diff
@@ -76,6 +78,52 @@ func (d *Docker) AddMount(typeStr, source, target string, optAdditionalArgs ...s
 	d.AddArgs("--mount", fmt.Sprintf("type=%s,source=%s,target=%s%s", typeStr, source, target, addl))
 }
 
+// CopyAndAddMount copies the source files into a temp directory, and then
+// mounts them as the target. It also returns a function to remove the temp
+// directory for cleaning up.
+func (d *Docker) CopyAndAddMount(typeStr, parentDir, source, target string, optAdditionalArgs ...string) func() {
+	fi, err := os.Stat(source)
+	if err != nil {
+		log.Fatalf("Error getting the FileInfo for %q: %v", source, err)
+		return nil
+	}
+
+	tempDir, err := ioutil.TempDir(parentDir, fi.Name())
+	if err != nil {
+		log.Fatal("error creating a temporary directory: ", err)
+		return nil
+	}
+
+	cleanup := func() {
+		log.Printf("🧹 Removing the temp directory %q", tempDir)
+		if err := os.RemoveAll(tempDir); err != nil {
+			log.Printf("Warning: error removing the temp directory %q: %v", tempDir, err)
+		}
+	}
```

Comment:
Being this a directory removal, why not making it an independent function and simply return the directory path as end result?
I'm not sure if this is a common practice in Go, but I'm not able to see the direct benefits of dealing with the function as a return object (it mainly complicates its interface)

Reply:
It's a common pattern in Go: a function returns another function that does the cleanup, and the callsite call `defer cleanup()` to make sure it's always being called, https://github.com/knative/serving/blob/master/cmd/queue/execprobe.go#L109 is one example and there are a lot more examples in other places.